step-capstone/src/components/Sidebars/InsertObjectPopover.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from "react";
+import { Grid, Popover, Box } from "@material-ui/core"
+import AddItemButton from "../TravelObjectForms/AddItemButton"
+import GetSuggestionButton from "../Suggestions/GetSuggestionButton"
+
+export default function InsertObjectPopover(props) {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+
+  useEffect(() => {
+    if (props.anchorEl !== anchorEl) {
+      setAnchorEl(props.anchorEl);
+    }
+  }, [props.anchorEl]);
+
+  const getCoordinates = () => {
+    let coordinates = { lat: 0, lng: 0 }
+    if (props.slots !== null) {
+      if (props.slots.prevTravelObject !== null && props.slots.nextTravelObject !== null && props.slots.prevTravelObject.type !== 'flight' && props.slots.nextTravelObject.type !== 'flight') {
```

Comment:
Nit: For super long criteria, it can be more readable if you pull out vars named after the  conditions as below. You can then re-use these for multiple criteria.

```
let prevNotNullOrFlight = props.slots.prevTravelObject !== null && props.slots.prevTravelObject.type !== 'flight'
let nextNotNullOrFlight = props.slots.nextTravelObject !== null && props.slots.nextTravelObject.type !== 'flight'
```

Reply:
done, great idea, thanks